python/lsst/summit/utils/consdbClient.py

```diff
@@ -0,0 +1,412 @@
+        """Insert values into a single ConsDB fixed metadata table.
+
+        Parameters
+        ----------
+        instrument: `str`
+            Name of the instrument (e.g. ``LATISS``).
+        table: `str`
+            Name of the table to insert into.
+        obs_id: `int`
+            Unique observation id.
+        values: `dict` [ `str`, `Any` ], optional
+            Dictionary of column/value pairs to add for the observation.
+        allow_update: `bool`, optional
+            If ``True``, allow replacement of values of existing columns.
+        **kwargs: `dict`
+            Additional column/value pairs, overriding ``values``.
+
+        Returns
+        -------
+        response: `requests.Response`
+            HTTP response from the server, with 200 status for success.
+        """
+        if values:
+            values.update(kwargs)
+        else:
+            values = kwargs
+        data = {"table": table, "obs_id": obs_id, "values": values}
+        url = _urljoin(
+            self.url,
+            "insert",
+            quote(instrument),
+            quote(table),
+            "obs",
+            quote(str(obs_id)),
+        )
+        if allow_update:
+            url += "?u=1"
+        return self._handle_post(url, data)
+
+    def query(self, query: str) -> Table:
+        """Query the ConsDB database.
+
+        Parameters
+        ----------
+        query: `str`
+            A SQL query (currently) to the database.
+
+        Returns
+        -------
+        result: `Table`
+            An ``astropy.Table`` containing the query results.
+
+        Notes
+        -----
+        This is a very general query interface because it is expected that
+        a wide variety of types of queries will be needed. If some types prove
+        to be common, syntactic sugar could be added to make the simpler.
+        """
+        url = _urljoin(self.url, "query")
+        data = {"query": query}
+        result = self._handle_post(url, data).json()
+        return Table(rows=result["data"], names=result["columns"])
+
+    def schema(self, instrument: str, table: str) -> list[tuple[str, str, str]]:
+        """Retrieve the schema of a fixed metadata table in ConsDB.
+
+        Parameters
+        ----------
+        instrument: `str`
+            Name of the instrument (e.g. ``LATISS``).
+        table: `str`
+            Name of the table to insert into.
+
+        Returns
+        -------
+        column_info: `list` [ `tuple` [ `str`, `str`, `str` ] ]
+            List of tuples, one for each column in the table. Each tuple
+            contains the name of the key, a data type string, and a
+            documentation string.
+
+        Notes
+        -----
+        Fixed metadata data types may use the full database vocabulary,
+        unlike flexible metadata data types.
+        """
+        url = _urljoin(self.url, "schema", quote(instrument), quote(table))
+        return self._handle_get(url)
```

Comment:
See comments on `get_flexible_metadata_keys()` about potentially having the same return shape.